drake/util/drakeGradientUtil.h

```diff
@@ -30,6 +31,31 @@ std::array<int, Size> intRange(int start)
   return ret;
 }
 
+namespace Drake {
+  // todo: recursive template to get arbitrary gradient order
+
+  // note: tried using template default values (e.g. Eigen::Dynamic), but they didn't seem to work on my mac clang
+  template <int num_vars> using TaylorVar = Eigen::AutoDiffScalar< Eigen::Matrix<double,num_vars,1> >;
+  template <int num_vars, int cols> using TaylorVec = Eigen::Matrix< TaylorVar<num_vars>, cols, 1>;
```

Comment:
Yeah, this would work. I don't have higher order yet. Should we worry about
that when we need them?

On Mon, Aug 31, 2015 at 10:17 PM, Russ Tedrake <notifications@github.com>
wrote:

> In drake/util/drakeGradientUtil.h
> <https://github.com/RobotLocomotion/drake/pull/1294#discussion_r38380262>:
>
> > @@ -30,6 +31,31 @@ std::array<int, Size> intRange(int start)
> >    return ret;
> >  }
> >
> > +namespace Drake {
> > +  // todo: recursive template to get arbitrary gradient order
> > +
> > +  // note: tried using template default values (e.g. Eigen::Dynamic), but they didn't seem to work on my mac clang
> > +  template <int num_vars> using TaylorVar = Eigen::AutoDiffScalar< Eigen::Matrix<double,num_vars,1> >;
> > +  template <int num_vars, int cols> using TaylorVec = Eigen::Matrix< TaylorVar<num_vars>, cols, 1>;
>
> good catch.
>
> are you happy with these defines? i'd like to generlize them for high
> orders (as the todo says), but thought you already have that somewhere. i'd
> like to agree (or not) on the TaylorVar types and make sure we use them in
> all this new code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RobotLocomotion/drake/pull/1294/files#r38380262>.
>